okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java

```diff
@@ -1417,13 +1417,13 @@ static void canonicalize(StringBuilder out, String input, int pos, int limit,
         utf8Buffer.writeUtf8CodePoint(codePoint);
         while (!utf8Buffer.exhausted()) {
           int b = utf8Buffer.readByte() & 0xff;
-          out.append('%');
-          out.append(HEX_DIGITS[(b >> 4) & 0xf]);
-          out.append(HEX_DIGITS[b & 0xf]);
+          out.writeByte('%');
+          out.writeByte(HEX_DIGITS[(b >> 4) & 0xf]);
+          out.writeByte(HEX_DIGITS[b & 0xf]);
         }
       } else {
         // This character doesn't need encoding. Just copy it over.
-        out.append((char) codePoint);
+        out.writeUtf8CodePoint((char) codePoint);
```

Comment:
Yeah, actually the code there now will be incorrect when I fix fragments to not encode non-ascii chars.